apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -13,25 +13,83 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
+/**
+ * ApolloCall is an abstraction for a request that has been prepared for execution. ApolloCall
+ * represents a single request/response pair and cannot be executed twice, though it can be cancelled.
+ */
 public interface ApolloCall<T> extends Cancelable {
 
+  /**
+   * Sends the request immediately and blocks until the response can be processed or is an error.
+   *
+   * @return The successful or failed {@link Response}
+   * @throws ApolloException       if the request could not be executed due to a cancellation, a timeout or a network
+   *                               failure
+   * @throws IllegalStateException when the call has already been executed
+   */
   @Nonnull Response<T> execute() throws ApolloException;
 
+  /**
+   * Schedules the request to be executed at some point in the future.
+   * The dispatcher defines when the request will run: usually immediately unless there are several other requests
+   * currently being executed.
```

Comment:
Not sure about the mentioned dispatcher, as it's not quite correct as we are actually using 2 dispatchers: 1st used to kickoff the request plus it will go to normalized cache, 2nd is okHttp3 dispatcher that is used internally with http call `okhttp3.Call#enqueue`